Backport 34ae47a22629 the real one

vrothberg opened the pull request:
* backport commit 34ae47a22629
* bump to v1.19.8
* bump to v1.19.9-dev

@nalind @rhatdan PTAL

I want that in https://github.com/containers/podman/pull/9630.  Vendoring master barks as the F33 images need an updated libcap.

vrothberg commented:
```
[+0241s] # github.com/containers/buildah/copier [github.com/containers/buildah/copier.test]
[+0241s] copier/copier_linux_test.go:114:10: t.TempDir undefined (type *testing.T has no field or method TempDir)
[+0241s] copier/copier_test.go:494:13: t.TempDir undefined (type *testing.T has no field or method TempDir)
[+0241s] copier/copier_test.go:533:13: t.TempDir undefined (type *testing.T has no field or method TempDir)
[+0241s] copier/copier_test.go:557:13: t.TempDir undefined (type *testing.T has no field or method TempDir)
[+0262s] === RUN   TestNoop
```

That's a method in the standard lib. Looks like the CI images in the `release-1.19` branch are (too) outdated.

I see some commits in main which bumped the images there but I am absolutely not sure if I can just cherry-pick them here.

@cevich PTAL

nalind commented:
> ```
> [+0241s] # github.com/containers/buildah/copier [github.com/containers/buildah/copier.test]
> [+0241s] copier/copier_linux_test.go:114:10: t.TempDir undefined (type *testing.T has no field or method TempDir)
> [+0241s] copier/copier_test.go:494:13: t.TempDir undefined (type *testing.T has no field or method TempDir)
> [+0241s] copier/copier_test.go:533:13: t.TempDir undefined (type *testing.T has no field or method TempDir)
> [+0241s] copier/copier_test.go:557:13: t.TempDir undefined (type *testing.T has no field or method TempDir)
> [+0262s] === RUN   TestNoop
> ```
> 
> That's a method in the standard lib. Looks like the CI images in the `release-1.19` branch are (too) outdated.

Whoops, those were introduced in Go 1.15 and I replaced some uses of `ioutil.TempDir()` with it without realizing.  Fedora 32 had 1.14, but Fedora 33 has 1.15.

TomSweeneyRedHat commented:
I don't think this should go into RHEL 8.4, and I know that @rhatdan is planning to vendor an updated Buildah into Podman for RHEl 8.4.  Do we need to think about creating a new branch for RHEL 8.4?  (shudder)

rhatdan commented:
Since this is just for the test, it probably is less important, correct.

vrothberg commented:
Please don't block copy fixes for 8.4.  It's almost impossible to untangle them and this is fixing an issue for rootless `podman cp` which I really want/need in 8.4 since other fixes are based on top.

vrothberg commented:
> Whoops, those were introduced in Go 1.15 and I replaced some uses of `ioutil.TempDir()` with it without realizing. Fedora 32 had 1.14, but Fedora 33 has 1.15.

Changed to use `ioutil.TempDir(...)`

rhatdan commented on buildah.go:
```diff
@@ -28,7 +28,7 @@ const (
 	// Version for the Package.  Bump version in contrib/rpm/buildah.spec
 	// too.
-	Version = "1.19.8"
+	Version = "1.19.9-dev"
```
I don't think this PR is necessary.  ON the supported branches we don't have -dev packages.

vrothberg replied:
Dropped the specific commit :+1:

rhatdan commented:
/lgtm

cevich commented:
> I see some commits in main which bumped the images there but I am absolutely not sure if I can just cherry-pick them here.

Most often this causes even more tests to fail, and it's not a very "kosher" thing to do on a release branch.  The branches are about locking environmental things down for stability.  Changing golang/glibc stuff under the covers, tends to break promises and cause baby-$DEITY to cry.
